core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -425,6 +426,8 @@ public ManagedChannelImpl shutdown() {
       maybeTerminateChannel();
       if (!terminated) {
         transportsCopy.addAll(transports.values());
+        transports.clear();
+        decommissionedTransports.addAll(transportsCopy);
```

Comment:
I wanted getTransport()'s `transports.get()` to return `null`, so that we end up going to the slow path and return `SHUTDOWN_TRANSPORT`.